proxy_list.go

```diff
@@ -0,0 +1,332 @@
+	if err != nil {
+		return 0, err
+	}
+	return codec.DecodeListLastIndexOfResponse(response), nil
+}
+
+// Remove removes the given element from this list.
+// Returns true if the list has changed as the result of this operation, false otherwise.
+func (l *List) Remove(element interface{}) (bool, error) {
+	elementData, err := l.validateAndSerialize(element)
+	if err != nil {
+		return false, err
+	}
+	request := codec.EncodeListRemoveRequest(l.name, elementData)
+	response, err := l.invokeOnPartition(context.TODO(), request, l.partitionID)
+	if err != nil {
+		return false, err
+	}
+	return codec.DecodeListRemoveResponse(response), nil
+}
+
+// RemoveAt removes the element at the given index.
+// Returns the removed element.
+func (l *List) RemoveAt(index int32) (interface{}, error) {
+	request := codec.EncodeListRemoveWithIndexRequest(l.name, index)
+	response, err := l.invokeOnPartition(context.TODO(), request, l.partitionID)
+	if err != nil {
+		return nil, err
+	}
+	return l.convertToObject(codec.DecodeListRemoveWithIndexResponse(response))
+}
+
+// RemoveAll removes the given elements from the list.
+// Returns true if the list has changed as the result of this operation, false otherwise.
+func (l *List) RemoveAll(elements ...interface{}) (bool, error) {
+	elementsData, err := l.validateAndSerializeValues(elements...)
+	if err != nil {
+		return false, err
+	}
+	request := codec.EncodeListCompareAndRemoveAllRequest(l.name, elementsData)
+	response, err := l.invokeOnPartition(context.TODO(), request, l.partitionID)
+	if err != nil {
+		return false, err
+	}
+	return codec.DecodeListCompareAndRemoveAllResponse(response), nil
+}
+
+// RetainAll removes all elements from this list except the ones contained in the given slice.
+// Returns true if the list has changed as a result of this operation, false otherwise.
+func (l *List) RetainAll(elements ...interface{}) (bool, error) {
+	elementsData, err := l.validateAndSerializeValues(elements...)
+	if err != nil {
+		return false, err
+	}
+	request := codec.EncodeListCompareAndRetainAllRequest(l.name, elementsData)
+	response, err := l.invokeOnPartition(context.TODO(), request, l.partitionID)
+	if err != nil {
+		return false, err
+	}
+	return codec.DecodeListCompareAndRetainAllResponse(response), nil
+}
+
+// Set replaces the element at the specified index in this list with the specified element.
+// Returns the previous element from the list.
+func (l *List) Set(index int32, element interface{}) (interface{}, error) {
+	elementData, err := l.validateAndSerialize(element)
+	if err != nil {
+		return nil, err
+	}
+	request := codec.EncodeListSetRequest(l.name, index, elementData)
+	response, err := l.invokeOnPartition(context.TODO(), request, l.partitionID)
+	if err != nil {
+		return nil, err
+	}
+	return l.convertToObject(codec.DecodeListSetResponse(response))
+}
+
+// Size returns the number of elements in this list.
+func (l *List) Size() (int32, error) {
```

Comment:
The same comment about `int32` vs `int` as the `index`, but for the return value...